test/tools/unified-spec-runner/match.ts

```diff
@@ -591,6 +633,30 @@ export function matchesEvents(
   }
 }
 
+export function filterLogs(
+  logsToIgnore: ExpectedLogMessage[],
+  actual: ExpectedLogMessage[],
+  entities: EntitiesMap
+): ExpectedLogMessage[] {
+  function isLogRelevant(log: ExpectedLogMessage) {
```

Comment:
This is the filter function to filter out any logs that the spec tests list as "IgnoreMessages" so we don't fail the equality check.

Reply:
Okay I followed up with the spec here, a few Qs:

- why check for level and component? shouldn't we be able to use resultCheck as a whole? (could be missing something)
  - What about the `failureIsRedacted` field?
- "which MUST be ignored on the corresponding client while executing [operations](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#test-operations)." This would imply that we need to invoke this whenever we go to collect a long, as opposed to filtering them at the end of the test run.

Reply:
@nbbeeken re your second point, the sentence that follows clarifies that the filtering just has to happen before we evaluate the `messages` directive, which could be after all logs have been collected or filtered by the logger's `logDestination`.

Relevant sentence:
> The test runner MUST exclude all log messages from observed messages that match any of the messages in `ignoreMessages` array before `messages` evaluation.

@aditi-khare-mongoDB One other follow up question:
- what's the expected behaviour when there's overlap in the `messages` array and the `ignoreMessages` array?